pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -350,7 +435,20 @@ func (p *staticPolicy) GetTopologyHints(s state.State, pod *v1.Pod, container *v
 		return nil
 	}
 
-	requestedResources, err := getRequestedResources(container)
+	var requestedResources map[v1.ResourceName]uint64
+	var err error
+
+	// Init containers should be placed on NUMA nodes where inheriting container's requirement can be satisfied
+	for _, initContainer := range pod.Spec.InitContainers {
+		if container.Name == initContainer.Name {
+			requestedResources, err = getMaxRequestedResources(pod)
```

Comment:
I think you should not run it for each init container, it should be enough to run it only once and save the value under some object variable